packages/cli/src/commands/init/index.ts

```
@@ -0,0 +1,177 @@
+        if (validation.valid) {
+          return true;
+        }
+
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        return 'Invalid project name: ' + (validation as any).problems[0];
+      },
+    });
+
+    if (typeof res.path === 'string') {
+      projectPath = res.path.trim();
+    }
+  }
+
+  if (!projectPath) {
+    console.log(
+      '\nPlease specify the project directory:\n' +
+        `  ${cyan(programName)} ${green('<project-directory>')}\n` +
+        'For example:\n' +
+        `  ${cyan(programName)} ${green('my-novu-app')}\n\n` +
+        `Run ${cyan(`${programName} --help`)} to see all options.`
+    );
+    process.exit(1);
+  }
+
+  const resolvedProjectPath = path.resolve(projectPath);
+  const projectName = path.basename(resolvedProjectPath);
+
+  const validation = validateNpmName(projectName);
+  if (!validation.valid) {
+    console.error(`Could not create a project called ${red(`"${projectName}"`)} because of npm naming restrictions:`);
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (validation as any).problems.forEach((problem: string) => console.error(`    ${red(bold('*'))} ${problem}`));
+    process.exit(1);
+  }
+
+  let userId: string;
+  // if no secret key is supplied set to empty string
+  if (!program.secretKey) {
+    program.secretKey = '';
+  } else {
+    try {
+      const response = await fetch(program.apiUrl + '/v1/users/me', {
+        headers: {
+          Authorization: `ApiKey ${program.secretKey}`,
+        },
+      });
+
+      if (!response.ok) {
+        throw new Error('Failed to fetch environment details');
+      }
+
+      const user = await response.json();
+      userId = user.data._id;
+
+      analytics.alias({
+        previousId: anonymousId,
+        userId: user.data._id,
+      });
+    } catch (error) {
+      console.error('Error fetching environment details:', error);
```

Should we throw an exception here as well? 

```suggestion
console.error('The API key you provided could not be found in our system. Please verify your key or contact support.');
```

```suggestion
console.error('The provided API key seems to be invalid. Please check your key and try again.');
```

---

Yes, was also thinking about it 🤔